pysb/simulator/base.py

```diff
@@ -735,15 +783,230 @@ def observables(self):
+    @property
+    def initials(self):
+        return self._initials
+
+    @property
+    def param_values(self):
+        return self._param_values
+
+    def save(self, filename, dataset_name=None, group_name=None,
+             append=False, include_obs_exprs=False):
+        """
+        Save a SimulationResult to a file (HDF5 format)
+
+        HDF5 is a hierarchical, binary storage format well suited to storing
+        matrix-like data. Our implementation requires the h5py package.
+
+        Each SimulationResult is treated as an HDF5 dataset, stored within a
+        group which is specific to a model. In this way, it is possible to save
+        multiple SimulationResults for a specific model.
+
+        A group is first created in the HDF file root (see group_name argument)
+        and an attribute "model" has a pickled version of the PySB model.
+        SimulationResult datasets are then stored within this group.
+
+        The file hierarchy under group_name/dataset_name/ then consists of
+        the following HDF5 gzip compressed HDF5 datasets: trajectories,
+        param_values, initials, tout, observables (optional) and expressions
+        (optional); and the following attributes:
+        simulator_class (pickled Class), simulator_kwargs (pickled dict),
+        squeeze (bool), simulations_per_param_set (int), pysb_version (str),
+        creation_date (ISO 8601 format).
+
+        Parameters
+        ----------
+        filename: str
+            Filename to which the data will be saved
+        dataset_name: str or None
+            Dataset name. If None, will default to the name of the simulator
+            class. If the dataset_name already exists within the group,
+            a ValueError is raised.
+        group_name: str or None
+            Group name. If None, will default to the name of the model.
+        append: bool
+            If False, raise IOError if the specified file already exists. If
+            True, append to existing file (or create if it doesn't exist).
+        include_obs_exprs: bool
+            Whether to save observables and expressions in the file or not.
+            If they are not included, they can be recreated from the model
+            and species trajectories when loaded back into PySB, but you may
+            wish to include them for use with external software, or if you
+            have complex expressions which take a long time to compute.
+        """
+        if h5py is None:
+            raise Exception('Please install the h5py package for this feature')
+
+        if self._y is None and not include_obs_exprs:
+            warn('This SimulationResult has no trajectories - '
+                 'you will need to set include_obs_exprs=True if '
+                 'you wish to save observables and expressions')
+
+        if group_name is None:
+            group_name = self._model.name
+        if dataset_name is None:
+            dataset_name = self.simulator_class.__name__
```

Comment:
I find this an odd default for the dataset name. How about something neutral like `'result'`?